pycytominer/cyto_utils/collate.py

```diff
@@ -0,0 +1,242 @@
+        The list of image feature groups to be used by add_image_features during aggregation
+    printtoscreen: bool, optional, default True
+        Whether or not to print output to the terminal
+    """
+
+    from pycytominer.cyto_utils.cells import SingleCells
+
+    # Set up directories (these need to be abspaths to keep from confusing makedirs later)
+    input_dir = os.path.abspath(
+        os.path.join(base_directory, "analysis", batch, plate, pipeline)
+    )
+    backend_dir = os.path.abspath(os.path.join(base_directory, "backend", batch, plate))
+    cache_backend_dir = os.path.abspath(os.path.join(temp, "backend", batch, plate))
+
+    aggregated_file = os.path.join(backend_dir, plate + ".csv")
+    backend_file = os.path.join(backend_dir, plate + ".sqlite")
+    cache_backend_file = os.path.join(cache_backend_dir, plate + ".sqlite")
+
+    if not aggregate_only:
+        if os.path.exists(cache_backend_file):
+            if not overwrite:
+                sys.exit(
+                    f"An SQLite file for {plate} already exists at {cache_backend_file} and overwrite is set to False. Terminating."
+                )
+            else:
+                os.remove(cache_backend_file)
+
+        for eachdir in [input_dir, backend_dir, cache_backend_dir]:
+            if not os.path.exists(eachdir):
+                os.makedirs(eachdir, exist_ok=True)
+
+        if remote:
+
+            remote_input_dir = os.path.join(remote, "analysis", batch, plate, pipeline)
+            remote_backend_file = os.path.join(
+                remote, "backend", batch, plate, plate + ".sqlite"
+            )
+            remote_aggregated_file = os.path.join(
+                remote, "backend", batch, plate, plate + ".csv"
+            )
+
+            sync_cmd = (
+                'aws s3 sync --exclude "*" --include "*/Cells.csv" --include "*/Nuclei.csv" --include "*/Cytoplasm.csv" --include "*/Image.csv" '
+                + remote_input_dir
+                + " "
+                + input_dir
+            )
+            if printtoscreen:
+                print(f"Downloading CSVs from {remote_input_dir} to {input_dir}")
+            run_check_errors(sync_cmd)
+
+        ingest_cmd = [
+            "cytominer-database",
+            "ingest",
+            input_dir,
+            "sqlite:///" + cache_backend_file,
+            "-c",
+            config,
+        ]
+        if not munge:
+            # munge is by default True in cytominer-database
+            ingest_cmd.append("--no-munge")
+        if printtoscreen:
+            print(f"Ingesting {input_dir}")
+        run_check_errors(ingest_cmd)
+
+        if column:
+            if print:
+                print(f"Adding a Metadata_Plate column based on column {column}")
+            alter_cmd = [
+                "sqlite3",
+                cache_backend_file,
+                "ALTER TABLE Image ADD COLUMN Metadata_Plate TEXT;",
```

Comment:
Would it be super annoying to abstract all `sqlite3` command building lists to a separate function? My sense is that doing so will help with maintainability, readability, and testing.

Reply:
Doing so might also enable us to move beyond just the three hardcoded compartments below (3 different calls to cells, cytoplasm, nuclei) - what happens if there are other compartments? (or fewer)?

Reply:
sounds good. Doesn't seem like it is worth the time then. I described this issue in #204

Any additional thoughts here? Feel free to resolve conversation if not